checks/raw/shell_download_validate.go

```diff
@@ -525,6 +564,14 @@ func isUnpinnedPipInstall(cmd []string) bool {
 		hasAdditionalArgs = true
 	}
 
+	// If is editable install, it's secure if package is from local source
+	// or from remote (VCS install) pinned by hash, and if dependencies are
+	// not installed.
+	// Example: `pip install --no-deps -e git+https://git.repo/some_pkg.git@da39a3ee5e6b4b0d3255bfef95601890afd80709`
+	if isEditableInstall {
+		return !hasNoDeps || !isPinnedEditableInstall
```

Comment:
hoe do the new options work with existing `hasAdditionalArgs`, `hasWheels`, etc? Can we return safely here or do we need to return after checking the other options first (code below)?

Reply:
I will double check the test cases to see if we need to rearrange the return line.

Reply:
The cases with `--require-hashes` and `<pkg>.whl` are working fine. Since `--require-hashes` and `-e` is not possible by pip install syntax, I'm not testing this case. Same thing for `-e` with `<pkg>.whl`.
Only case I caught which is NOT working is support installing 2 packages with editable `-e` install, should we cover that?

Reply:
Let's cover it, yes. Otherwise someone may tell us we have false positives.
Can you also add comments about the reasoning you explain in your comment? That will also help document the code.

Thanks again for the hard work figuring out all these subtle differences.

Reply:
@laurentsimon Done. Please check if you agree.

Reply:
lgtm. I think the unit tests speak for themselves. If you've added some of these corner cases, we're good to go.